source/fab/tools/compiler_wrapper.py

```diff
@@ -0,0 +1,206 @@
+                               f"'set_module_output_path' function.")
+        cast(FortranCompiler, self._compiler).set_module_output_path(path)
+
+    def compile_file(self, input_file: Path,
+                     output_file: Path,
+                     openmp: bool,
+                     add_flags: Union[None, List[str]] = None,
+                     syntax_only: Optional[bool] = None):
+        # pylint: disable=too-many-arguments
+        '''Compiles a file using the wrapper compiler. It will temporarily
+        change the executable name of the wrapped compiler, and then calls
+        the original compiler (to get all its parameters)
+
+        :param input_file: the name of the input file.
+        :param output_file: the name of the output file.
+        :param openmp: if compilation should be done with OpenMP.
+        :param add_flags: additional flags for the compiler.
+        :param syntax_only: if set, the compiler will only do
+            a syntax check
+        '''
+
+        orig_compiler_name = self._compiler.exec_name
+        self._compiler.change_exec_name(self.exec_name)
+        if add_flags is None:
+            add_flags = []
+        if self._compiler.category is Category.FORTRAN_COMPILER:
+            # Mypy complains that self._compiler does not take the syntax
+            # only parameter. Since we know it's a FortranCompiler.
+            # do a cast to tell mypy that this is now a Fortran compiler
+            # (or a CompilerWrapper in case of nested CompilerWrappers,
+            # which also supports the syntax_only flag anyway)
+            self._compiler = cast(FortranCompiler, self._compiler)
+            self._compiler.compile_file(input_file, output_file, openmp=openmp,
+                                        add_flags=self.flags + add_flags,
+                                        syntax_only=syntax_only,
+                                        )
+        else:
+            if syntax_only is not None:
+                raise RuntimeError(f"Syntax-only cannot be used with compiler "
+                                   f"'{self.name}'.")
+            self._compiler.compile_file(input_file, output_file, openmp=openmp,
+                                        add_flags=self.flags+add_flags
+                                        )
+        self._compiler.change_exec_name(orig_compiler_name)
+
+
+# ============================================================================
+class Mpif90(CompilerWrapper):
+    '''Class for a simple wrapper for using a compiler driver (like mpif90)
+    It will be using the name "mpif90-COMPILER_NAME" and calls `mpif90`.
+    All flags from the original compiler will be used when using the wrapper
+    as compiler.
+
+    :param compiler: the compiler that the mpif90 wrapper will use.
+    '''
+
+    def __init__(self, compiler: Compiler):
+        super().__init__(name=f"mpif90-{compiler.name}",
+                         exec_name="mpif90", compiler=compiler, mpi=True)
```

Comment:
I'm concerned that we end up with a combinatorial problem here. For example when we start thinking about profilers, some of which wrap the compiler, we could end up with objects for compiler, MPI compiler, profiled compiler and profiled MPI compiler.

Can we wrap the compiler dynamically, rather than holding it named in the tool kit, and perform a test in the constructor to ensure there is actually an MPI built with that compiler at that point?